File: catkin_ws/src/beginner_tutorials/src/talker.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

int main(int argc, char **argv)
{
    //ros::init(argc, argv, [node name])
    ros::init(argc, argv, "talker"); 

    //node handle is to initialize node so that the node can communicate within ROS system
    ros::NodeHandle nh; 

    ros::Publisher chatter_pub = nh.advertise<std_msgs::String>("chatter", 1000); //([topic name], queue)

    ros::Rate loop_rate(10); //10 Hz

    int count = 0; //counter how many messages that have been sent

    while (ros::ok())
    {
        std_msgs::String msg;

        std::stringstream ss;
        ss<<"hello world"<<count;
        msg.data = ss.str();
        ROS_INFO("%s", msg.data.c_str());
        chatter_pub.publish(msg);
        
        ros::spinOnce();

        loop_rate.sleep();
        count++;
    }
    
    return 0;
}

File: catkin_ws/src/beginner_tutorials/scripts/talker.py
#!/usr/bin/env python
# license removed for brevity
import rospy
from std_msgs.msg import String

def talker():
    rospy.init_node('talker', anonymous=True) #(node name, ensure unique)
    pub = rospy.Publisher('chatter', String, queue_size = 10) #(topic name, data type, queue size)
    rate = rospy.Rate(10) #10Hz
    while not rospy.is_shutdown(): 
        hello_str = "hello world %s" % rospy.get_time()
        rospy.loginfo(hello_str) #std::cout or printf
        pub.publish(hello_str)
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException: #SIGINT handler
        pass